test_download_specific_git_annex_release_asset on windows downloads incomplete file without error

happened in one recent run https://github.com/datalad/datalad-installer/actions/runs/4351210903/jobs/7602645257 where the other matrix runs were all good. I would have expected some error instead of some incomplete download
```
================================== FAILURES ===================================
_ test_download_specific_git_annex_release_asset[ubuntu-8.20211231-git-annex-standalone_8.20211231-1.ndall+1_amd64.deb-49966540] _

ostype = 'ubuntu', version = '8.20211231'
filename = 'git-annex-standalone_8.20211231-1.ndall+1_amd64.deb'
size = 49966540
tmp_path = WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-unknown/pytest-0/test_download_specific_git_ann0')

    @pytest.mark.ghauth
    @pytest.mark.parametrize(
        "ostype,version,filename,size",
        [
            (
                "ubuntu",
                "8.20211231",
                "git-annex-standalone_8.20211231-1.ndall+1_amd64.deb",
                49966540,
            ),
            ("macos", "8.20211123", "git-annex_8.20211123_x64.dmg", 28809446),
            ("windows", "8.20211117", "git-annex-installer_8.20211117_x64.exe", 18156853),
        ],
    )
    def test_download_specific_git_annex_release_asset(
        ostype: str, version: str, filename: str, size: int, tmp_path: Path
    ) -> None:
        DataladGitAnnexReleaseBuildInstaller.download(
            ostype=ostype,
            target_dir=tmp_path,
            version=version,
        )
        (p,) = tmp_path.iterdir()
        assert p.is_file()
        assert p.name == filename
>       assert p.stat().st_size == size
E       AssertionError: assert 45[154](https://github.com/datalad/datalad-installer/actions/runs/4351210903/jobs/7602645257#step:5:155)03 == 49966540
E        +  where 4515403 = os.stat_result(st_mode=33206, st_ino=1125899908260822, st_dev=1879469748, st_nlink=1, st_uid=0, st_gid=0, st_size=4515403, st_atime=[167](https://github.com/datalad/datalad-installer/actions/runs/4351210903/jobs/7602645257#step:5:168)8[170](https://github.com/datalad/datalad-installer/actions/runs/4351210903/jobs/7602645257#step:5:171)736, st_mtime=1678170736, st_ctime=1678170629).st_size
E        +    where os.stat_result(st_mode=33206, st_ino=1125899908260822, st_dev=1879469748, st_nlink=1, st_uid=0, st_gid=0, st_size=4515403, st_atime=1678170736, st_mtime=1678170736, st_ctime=1678170629) = <bound method Path.stat of WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-unknown/pytest-0/test_download_specific_git_ann0/git-annex-standalone_8.20211231-1.ndall+1_amd64.deb')>()
E        +      where <bound method Path.stat of WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-unknown/pytest-0/test_download_specific_git_ann0/git-annex-standalone_8.20211231-1.ndall+1_amd64.deb')> = WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-unknown/pytest-0/test_download_specific_git_ann0/git-annex-standalone_8.20211231-1.ndall+1_amd64.deb').stat

test\test_install.py:490: AssertionError
------------------------------ Captured log call ------------------------------
DEBUG    datalad_installer:datalad_installer.py:2342 HTTP request: GET https://api.github.com/repos/datalad/git-annex/releases/tags/8.20211231
INFO     datalad_installer:datalad_installer.py:2538 Downloading https://github.com/datalad/git-annex/releases/download/8.20211231/git-annex-standalone_8.20211231-1.ndall%2B1_amd64.deb
=========================== short test summary info ===========================
FAILED test/test_install.py::test_download_specific_git_annex_release_asset[ubuntu-8.20211231-git-annex-standalone_8.20211231-1.ndall+1_amd64.deb-49966540]
====== 1 failed, 92 passed, 10 skipped, 1 xpassed in 1304.24s (0:21:44) =======
py: exit 1 (1305.80 seconds) D:\a\datalad-installer\datalad-installer> pytest -vv --ci --cov-report=xml test pid=1188
.pkg: _exit> python C:\hostedtoolcache\windows\Python\3.11.2\x64\Lib\site-packages\pyproject_api\_backend.py True setuptools.build_meta
  py: FAIL code 1 (1332.36=setup[26.56]+cmd[1305.80] seconds)
  evaluation failed :( (1332.67 seconds)
Error: Process completed with exit code 1.
```

Making a record to see if we run into it again.

--- a/src/datalad_installer.py
+++ b/src/datalad_installer.py
@@ -2545,6 +2545,13 @@ def download_file(
             with urlopen(req) as r:
                 with open(path, "wb") as fp:
                     shutil.copyfileobj(r, fp)
+                if "content-length" in r.headers:
+                    size = int(r.headers["Content-Length"])
+                    fsize = os.path.getsize(path)
+                    if fsize < size:
+                        raise URLError(
+                            f"only {fsize} out of {size} bytes were received"
+                        )
             return
         except URLError as e:
             if isinstance(e, HTTPError) and e.code not in (500, 502, 503, 504):

